### PR TITLE
add additional workload-identity role-bindings

### PR DESCRIPTION
**Pull request by rob-1126**
One of our VIP customers wanted more information about the service-accounts and a listing of all those that might need xcom access. This updates our previously incomplete list and adds a bit more detail to make this VIP happy.

### CODE REVIEW

**Comment by rob-1126** [comment]
resuscitates PR 3862

File: software/integrate-iam.md [hunk 1]
```diff
@@ -326,3 +327,15 @@ astronomer:
     ```
 
     If Workload Identity is working, you should see a list of credentialed accounts related to your GCP service account.
+
+## Granting Access to Airflow components using XCom and secrets backends {sas-that-use-backends}
```
Comment (lzdanski) [inline software/integrate-iam.md:331]:
```suggestion
## Granting Access to Airflow components using XCom and secrets backends {#sas-that-use-backends}
```

File: software/integrate-iam.md [hunk 2]
```diff
@@ -326,3 +327,15 @@ astronomer:
     ```
 
     If Workload Identity is working, you should see a list of credentialed accounts related to your GCP service account.
+
+## Granting Access to Airflow components using XCom and secrets backends {#sas-that-use-backends}
+Astronomer Software creates a set of service-accounts for each Airflow instance it manages. The following roles can require access to Airflow Xcom backends or secrets backends to function:
```
Comment (lzdanski) [inline software/integrate-iam.md:332]:
```suggestion
Astronomer Software creates a set of service accounts for each Airflow instance it manages. The following roles can require access to Airflow Xcom backends or secrets backends to function:
```

File: software/integrate-iam.md [hunk 3]
```diff
@@ -326,3 +327,15 @@ astronomer:
...
+
+| Component                   | Rationale                                                                                                 |
+|-----------------------------|-----------------------------------------------------------------------------------------------------------|
+| `<release name>-scheduler`         | User-provided DAG code is regularly interpreted to determine what tasks are part of DAGs, and that code might incorporate references to values stored in secret storage or XCom.                  |
+| `<release name>-worker`            | User-provided DAG code is interpreted at task-run time and might incorporate references to values stored in secret storage or XCom.             |
+| `<release name>-triggerer`         | User-provided DAG code is interpreted at task run time and might incorporate references to values stored in secrets storage or XCom.            |
```
Comment (lzdanski) [inline software/integrate-iam.md:338]: "user provided code is interpreted" by Astronomer Software? By Airflow? What is doing the interpreting for each of these?

File: software/integrate-iam.md [hunk 4]
```diff
@@ -326,3 +327,15 @@ astronomer:
...
+| `<release name>-webserver`         | Provides users a mechanism to view and set secrets and to view XCom entries.|
```
Comment (lzdanski) [inline software/integrate-iam.md:339]:
```suggestion
| `<release name>-webserver`         | Provides a mechanism for you to view XCom entries and to view and set secrets. |
```

Again, who is the user?

File: software/integrate-iam.md [hunk 5]
```diff
@@ -326,3 +327,15 @@ astronomer:
...
+| `<release name>-cleanup`           |  Accesses task instances and other data that might include serialized references to values stored in secrets backends or XCOM backends.       |
```
Comment (lzdanski) [inline software/integrate-iam.md:340]: The `_ -cleanup` service account accesses task instances..?

**Review by lzdanski — CHANGES_REQUESTED** [review]
Some questions for you about what components/software is doing which actions?

File: software/integrate-iam.md [hunk 6]
```diff
@@ -326,3 +327,15 @@ astronomer:
...
+Astronomer Software creates a set of service accounts for each Airflow instance it manages. The following roles can require access to Airflow Xcom backends or secrets backends to function:
+
+| Component                   | Rationale                                                                                                 |
+|-----------------------------|-----------------------------------------------------------------------------------------------------------|
+| `<release name>-scheduler`         | User-provided DAG code is regularly interpreted by the scheduler to determine what tasks are part of DAGs, and that code might incorporate references to values stored in secret storage or XCom.                  |
+| `<release name>-worker`            | User-provided DAG code is interpreted by the worker at task run-time and might incorporate references to values stored in secret storage or XCom.             |
+| `<release name>-triggerer`         | User-provided DAG code is interpreted by the triggerer at task run-time and might incorporate references to values stored in secrets storage or XCom.            |
```
Comment (lzdanski) [inline software/integrate-iam.md:338]:
```suggestion
| `<release name>-scheduler`         | Regularly interprets your DAG code by the scheduler to determine which tasks are part of DAGs and might incorporate references to values stored in secret storage or XCom.                  |
| `<release name>-worker`            | Regularly interprets your DAG code by the worker at task runtime and might incorporate references to values stored in secret storage or XCom.             |
| `<release name>-triggerer`         | Regularly interprets your DAG code by the triggerer at task run-time and might incorporate references to values stored in secrets storage or XCom.            |
```

is this correct?

**Review by lzdanski — CHANGES_REQUESTED** [review]

**Review by lzdanski — APPROVED** [review]

**Pull request merged**